src/Sulu/Bundle/AdminBundle/Admin/View/TabViewBuilder.php

```diff
@@ -26,4 +26,17 @@ public function getView(): View
     {
         return clone $this->view;
     }
+
+    public function addRouterAttributesToBlacklist(
```

Comment:
Should we extract this method into a trait? Unfortunately, the `TabViewBuilderTrait` is already used by children of a tab view.

Reply:
Still not a big fan of traits :see_no_evil: But as explained above, the current situation makes that necessary. So yeah, we should extract that into a trait. Renaming the other trait would be a BC break... Any suggestions for a new name? Maybe we could even use the same trait, since the methods are private anyway, but that feels a bit weird...

@sulu/core-team WDYT?

Reply:
Could we maybe rename the current `TabViewBuilderTrait` to e.g. `ChildTabViewBuilderTrait`, create a new `TabViewBuilderTrait` with `addRouterAttributesToBlacklist` and use `ChildTabViewBuilderTrait` there. Then mark the methods from `ChildTabViewBuilderTrait` inside of `TabViewBuilderTrait` as deprecated. Would that solve the bc break issue?

Reply:
Example:

~~~php
trait TabViewBuilderTrait
{
    use ChildTabViewBuilderTrait {
        setTabTitleToView as parentSetTabTitleToView;
        setTabConditionToView as parentSetTabConditionToView;
        setTabOrderToView as parentSetTabOrderToView;
        setTabPriorityToView as parentSetTabPriorityToView;
    }

    /**
     * @deprecated Will be removed in Sulu 3.0. Use ChildTabViewBuilderTrait instead.
     */
    private function setTabTitleToView(View $view, string $tabTitle): void
    {
        $this->parentSetTabTitleToView($view, $tabTitle);
    }

    /**
     * @deprecated Will be removed in Sulu 3.0. Use ChildTabViewBuilderTrait instead.
     */
    private function setTabConditionToView(View $view, string $tabCondition): void
    {
        $this->parentSetTabConditionToView($view, $tabCondition);
    }

    /**
     * @deprecated Will be removed in Sulu 3.0. Use ChildTabViewBuilderTrait instead.
     */
    private function setTabOrderToView(View $view, int $tabOrder): void
    {
        $this->parentSetTabOrderToView($view, $tabOrder);
    }

    /**
     * @deprecated Will be removed in Sulu 3.0. Use ChildTabViewBuilderTrait instead.
     */
    private function setTabPriorityToView(View $view, int $tabPriority): void
    {
        $this->parentSetTabPriorityToView($view, $tabPriority);
    }
    
    public function addRouterAttributesToBlacklist(
        array $routerAttributesToBlacklist
    ): TabViewBuilderInterface {
        $oldRouterAttributesToBlacklist = $this->view->getOption('routerAttributesToBlacklist');
        $newRouterAttributesToBlacklist = $oldRouterAttributesToBlacklist
            ? array_merge($oldRouterAttributesToBlacklist, $routerAttributesToBlacklist)
            : $routerAttributesToBlacklist;

        $this->view->setOption('routerAttributesToBlacklist', $newRouterAttributesToBlacklist);

        return $this;
    }
}

trait ChildTabViewBuilderTrait
{
    private function setTabTitleToView(View $view, string $tabTitle): void
    {
        $view->setOption('tabTitle', $tabTitle);
    }

    private function setTabConditionToView(View $view, string $tabCondition): void
    {
        $view->setOption('tabCondition', $tabCondition);
    }

    private function setTabOrderToView(View $view, int $tabOrder): void
    {
        $view->setOption('tabOrder', $tabOrder);
    }

    private function setTabPriorityToView(View $view, int $tabPriority): void
    {
        $view->setOption('tabPriority', $tabPriority);
    }
}
~~~

Reply:
Would work for me. With a small addition: Every function being deprecated must trigger a deprecation error, otherwise Symfony does not recognize it as deprecation:

```php
@trigger_error('error message', E_USER_DEPRECATED);
```

Reply:
Are you fine with the name `ChildTabViewBuilderTrait`? I'm not very happy with it, it was just the first thing that came to my mind.

Reply:
When asking me like that I would maybe turn the words: `TabViewChildBuilderTrait`, so that the `Child` is at the end. Having the `TabView` at the end would probably mean that it is a `TabView` (at least we are following this convention in other places).

Reply:
fixed it